### PR TITLE
Implement vm_throw callback

### PR DESCRIPTION
**Pull request by zherczeg**
Slightly improve the description of vm_exec_stop callback.

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -11320,8 +11347,7 @@ backtrace_callback (jerry_backtrace_frame_t *frame_p,
 
 **Summary**
 
-When JERRY_FEATURE_VM_EXEC_STOP is enabled a callback function can be
-specified by this function. This callback is periodically called when
+The callback passed to this function is periodically called when
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md:11350]: How is this change related to the new vm throw? AFAIK these changes should not be in this PR, but in a separate one.

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -11393,12 +11424,77 @@ main (void)
...
+```c
+void
+jerry_set_vm_throw_callback (jerry_vm_throw_callback_t throw_cb,
+                             void *user_p);
+```
+
+- `throw_cb` - callback which is called on throws (passing NULL disables this feature)
+- `user_p` - user pointer passed to the `throw_cb` function
+
+*New in version 2.0*.
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md:11453]: This version info is incorrect, as it should be `[[NEXT_RELEASE]]`

File: docs/02.API-REFERENCE.md [hunk 3]
```diff
@@ -11393,12 +11424,77 @@ main (void)
...
+  const jerry_char_t script[] = "try { throw new Error('1') } catch (e) { throw new Error('2') }";
+
+  jerry_value_t parsed_code = jerry_parse (script, sizeof (script) - 1, NULL);
+  jerry_release_value (jerry_run (parsed_code));
+  jerry_release_value (parsed_code);
+
+  /* The counter contains 2. */
+
+  jerry_cleanup ();
+}
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md:11491]: There should be a `return 0` as the `main` expects an `int` return value.

File: docs/02.API-REFERENCE.md [hunk 4]
```diff
@@ -11393,12 +11424,77 @@ main (void)
...
+
+```c
+#include "jerryscript.h"
+
+static void
+vm_throw_callback (const jerry_value_t error_value, /**< captured error */
+                   void *user_p) /**< user pointer */
+{
+  (void) error_value;
+  (void) user_p;
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md:11467]: No need for this line.

File: docs/02.API-REFERENCE.md (hunk 1 above)
Comment (zherczeg) [reply docs/02.API-REFERENCE.md:11350]: I think it is not worth it. Small QOL changes.

**Comment by zherczeg** [comment]
Thank you for the review, patch is updated.

**Review by galpeter — APPROVED** [review]
lgtm

File: jerry-core/jcontext/jcontext.c [hunk 5]
```diff
@@ -113,8 +113,14 @@ jcontext_take_exception (void)
...
-  jcontext_set_abort_flag (false);
-  jcontext_set_exception_flag (false);
+#if JERRY_VM_THROW
+  JERRY_CONTEXT (status_flags) &= (uint32_t) ~(ECMA_STATUS_EXCEPTION
+                                               | ECMA_STATUS_ABORT
+                                               | ECMA_STATUS_ERROR_THROWN);
+#else /* !JERRY_VM_THROW */
+  JERRY_CONTEXT (status_flags) &= (uint32_t) ~(ECMA_STATUS_EXCEPTION
+                                               | ECMA_STATUS_ABORT);
+#endif /* JERRY_VM_THROW */
```
Comment (rerobika) [inline jerry-core/jcontext/jcontext.c:123]: How about this?

```c
  JERRY_CONTEXT (status_flags) &= (uint32_t) ~(ECMA_STATUS_EXCEPTION
  #if JERRY_VM_THROW
                                               | ECMA_STATUS_ERROR_THROWN
  #endif /* JERRY_VM_THROW */
                                               | ECMA_STATUS_ABORT);
```

**Review by rerobika — APPROVED** [review]
